samples/caf_sensor_manager/src/modules/workload_sim.c

```diff
@@ -0,0 +1,109 @@
+/*
+ * Copyright (c) 2022 Nordic Semiconductor ASA
+ *
+ * SPDX-License-Identifier: LicenseRef-Nordic-5-Clause
+ */
+
+#define MODULE workload_sim
+#include <stdlib.h>
+#include <caf/events/module_state_event.h>
+#include <caf/events/sensor_data_aggregator_event.h>
+
+#define BUSY_TIME CONFIG_WORKLOAD_TIME_PER_SAMPLE
+#define WORKLOAD_SIMULATED_THREAD_STACK_SIZE 800
+#define WORKLOAD_SIMULATED_THREAD_PRIORITY 1
+#define NUMBER_OF_WORKLOADS DT_PROP(DT_NODELABEL(agg0), buf_count)
+
+LOG_MODULE_REGISTER(MODULE);
+struct workload {
+	const struct sensor_data_aggregator_event *event;
+	struct k_work work;
+	bool busy;
+};
+
+static K_THREAD_STACK_DEFINE(workload_simulated_thread_stack,
+			     WORKLOAD_SIMULATED_THREAD_STACK_SIZE);
+static struct k_work_q workload_simulated_work_q;
+
+static struct workload workloads[NUMBER_OF_WORKLOADS];
+
+static void simulated_work_handler(struct k_work *work)
+{
+	struct workload *workload = CONTAINER_OF(work, struct workload, work);
+
+	/* Simulate processing of received data. */
+	k_busy_wait(BUSY_TIME * workload->event->sample_cnt);
```

Comment:
this may not work correctly for multiple instances (why do we need multiple instances - this sample uses only one aggregator and we don't plan to add more).

busy wait until CPU has done specific number of cycles (in this or other context)

Reply:
Ok, because only 1 instance runs at the time, they share common thread.